Improvement/896 retrieve service ip from pillar

ramanduh opened the pull request:
Retrieve services IP/ports from pillar metalk8s:endpoints in all sls files. If the pillar is absent an error is raised. As a consequence, we pass the pillar manually when the kubernetes object services which provide the endpoints data are not ready yet

Commit:
Refactor pillar use

Use metalk8s.endpoints.* pillar values when querying ip address of
registry, repository or salt master

Closes #896

bert-e commented:
# Hello ramanduh,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/896-retrieve-service-ip-from-pillar` into
`development/2.0`
* [w/2.1/improvement/896-retrieve-service-ip-from-pillar](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/896-retrieve-service-ip-from-pillar) will be merged into `development/2.1` 
* [w/2.2/improvement/896-retrieve-service-ip-from-pillar](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/896-retrieve-service-ip-from-pillar) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

NicolasT commented on salt/metalk8s/kubernetes/apiserver/installed.sls:
```diff
@@ -108,7 +108,7 @@ Create kube-apiserver Pod manifest:
           - --etcd-certfile=/etc/kubernetes/pki/apiserver-etcd-client.crt
           - --etcd-keyfile=/etc/kubernetes/pki/apiserver-etcd-client.key
-          - --etcd-servers=https://{{ pillar.registry_ip }}:2379
+          - --etcd-servers=https://{{ grains.metalk8s.control_plane_ip }}:2379
```
This may work now (and so does the old approach), but we'll need to change this at some point in time to point to the actual etcd cluster addresses. Could you file a tracking ticket?

ramanduh replied:
Yep, issue created: https://github.com/scality/metalk8s/issues/1074

NicolasT commented on salt/metalk8s/orchestrate/bootstrap/init.sls:
```diff
@@ -69,6 +77,7 @@ Deploy CA role on bootstrap minion:
   - sls:
     - metalk8s.roles.ca
   - saltenv: metalk8s-{{ version }}
+  - pillar: {{ pillar_data | tojson }}
```
Why?  The CA role should be fully stand-alone, no need to have any pillar values. Or am I missing something?

ramanduh replied:
It was failing when rendering `salt/metalk8s/repo/offline.sls`, the pillar `pillar.metalk8s.endpoints['package-repositories'].ip` was needed

NicolasT commented on salt/metalk8s/orchestrate/deploy_new_node.sls:
```diff
@@ -8,7 +8,6 @@
     'repo': {
```
`repo` should be gone as well, no?

ramanduh replied:
yes, will fix

NicolasT commented on salt/metalk8s/registry/installed.sls:
```diff
@@ -47,6 +47,6 @@ Install OCI registry manifest:
         registry_image: {{ registry_image }}
         registry_version: {{ registry_version }}
-        registry_ip: {{ pillar.registry_ip }}
+        registry_ip: {{ metalk8s.endpoints.registry.ip }}
```
Hmh, no. This was wrong in the beginning as well.

This should be set to the `control_plane_ip` (grain) of the node on which the service is deployed.

`metalk8s.endpoints.*` is a *result* of a running service, not the *input* to configure a service.

ramanduh replied:
I will update it to control_plane_ip

NicolasT commented on salt/metalk8s/repo/deployed.sls:
```diff
@@ -22,7 +20,7 @@ Deploy repo service object:
         ports:
         - name: http
-          port: {{ repo.port }}
+          port: {{ pillar.metalk8s.endpoints['package-repositories'].ports.http }}
```
Similar to the registry service IP before, this should be a fixed value.

ramanduh replied:
For the port, what do you mean by fixed value? Hardcoding it or letting it as it was before? (ie configurable via the default.yaml). My idea here is to force the caller of the state to specify the ports in the pillar arg if it is not available via the endpoints yet

NicolasT commented on salt/metalk8s/repo/installed.sls:
```diff
@@ -18,7 +18,7 @@ Generate package repositories nginx configuration:
     - backup: false
     - defaults:
-        listening_port: {{ repo.port }}
+        listening_port: {{ pillar.metalk8s.endpoints['package-repositories'].ports.http }}
```
Same as above.

NicolasT commented on salt/metalk8s/repo/installed.sls:
```diff
@@ -31,7 +31,7 @@ Install package repositories manifest:
     - backup: false
     - defaults:
-        container_port: {{ repo.port }}
+        container_port: {{ pillar.metalk8s.endpoints['package-repositories'].ports.http }}
```
Same as above.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

Commit:
Do not use pillar when configuring service

As at the moment we set up the service, the pillar shouldn't be
available yet

Relates #896

NicolasT commented:
@ramanduh Any updates? This seems to be the last PR we targeted for *alpha3*...

bert-e commented:
# History mismatch



Merge commit #da26cf8515c6b9221108cbfe87a325e74c33a274 on the integration branch
`w/2.1/improvement/896-retrieve-service-ip-from-pillar` is merging a branch which is neither the current
branch `improvement/896-retrieve-service-ip-from-pillar` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/896-retrieve-service-ip-from-pillar` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

ramanduh commented:
> @ramanduh Any updates? This seems to be the last PR we targeted for _alpha3_...

@NicolasT PR is ready for review again. Just relate to your previous review about repo.port
https://github.com/scality/metalk8s/pull/1062#discussion_r281169148

I let it customizable in default.yaml like before

ramanduh commented:
@bert-e approve

bert-e commented:
# History mismatch



Merge commit #da26cf8515c6b9221108cbfe87a325e74c33a274 on the integration branch
`w/2.1/improvement/896-retrieve-service-ip-from-pillar` is merging a branch which is neither the current
branch `improvement/896-retrieve-service-ip-from-pillar` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/896-retrieve-service-ip-from-pillar` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

ramanduh commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/896-retrieve-service-ip-from-pillar` into
`development/2.0`
* [w/2.1/improvement/896-retrieve-service-ip-from-pillar](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/896-retrieve-service-ip-from-pillar) will be merged into `development/2.1` 
* [w/2.2/improvement/896-retrieve-service-ip-from-pillar](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/896-retrieve-service-ip-from-pillar) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye ramanduh.
